### PR TITLE
BSD code cleanups.

### PR DESCRIPTION
**Pull request by devnexen**
Dead code/proper debug macros types.

### CODE REVIEW

File: libr/io/p/io_ptrace.c [hunk 1]
```diff
@@ -43,8 +43,13 @@ static int __waitpid(int pid) {
 	return (waitpid (pid, &st, 0) != -1);
 }
 
+#if defined(__GLIBC__) && defined(__linux__)
 #define debug_read_raw(io,x,y) r_io_ptrace((io), PTRACE_PEEKTEXT, (x), (void *)(y), NULL)
 #define debug_write_raw(io,x,y,z) r_io_ptrace((io), PTRACE_POKEDATA, (x), (void *)(y), (void *)(z))
+#else
+#define debug_read_raw(io,x,y) r_io_ptrace((io), PTRACE_PEEKTEXT, (x), (void *)(y), 0)
+#define debug_write_raw(io,x,y,z) r_io_ptrace((io), PTRACE_POKEDATA, (x), (void *)(y), (int)(z))
```
Comment (ret2libc) [inline libr/io/p/io_ptrace.c]: what about always using `r_ptrace_data_t`?

File: libr/io/p/io_ptrace.c [hunk 2]
```diff
@@ -43,8 +43,13 @@ static int __waitpid(int pid) {
 	return (waitpid (pid, &st, 0) != -1);
 }
 
+#if defined(__GLIBC__) && defined(__linux__)
 #define debug_read_raw(io,x,y) r_io_ptrace((io), PTRACE_PEEKTEXT, (x), (void *)(y), NULL)
 #define debug_write_raw(io,x,y,z) r_io_ptrace((io), PTRACE_POKEDATA, (x), (void *)(y), (void *)(z))
+#else
+#define debug_read_raw(io,x,y) r_io_ptrace((io), PTRACE_PEEKTEXT, (x), (void *)(y), 0)
```
Comment (ret2libc) [inline libr/io/p/io_ptrace.c]: is there no `NULL` on BSD?

File: libr/io/p/io_ptrace.c (hunk 1 above)
Comment (devnexen) [reply libr/io/p/io_ptrace.c]: Sure ... the outcome will be the same just little code reduction.

File: libr/io/p/io_ptrace.c (hunk 2 above)
Comment (devnexen) [reply libr/io/p/io_ptrace.c]: There is but as you pointed out there is r_ptrace_data_t and they have different underlying types Linux and non Linux for a reason.

Comment (ret2libc) [reply libr/io/p/io_ptrace.c]: I see. Then you could use the already existing `#if defined...` in r_io.h and do something like:
```C
#if defined(__GLIBC__) && defined(__linux__)
typedef enum __ptrace_request r_ptrace_request_t;
typedef void * r_ptrace_data_t;
#define R_PTRACE_NODATA NULL
#else
typedef int r_ptrace_request_t;
typedef int r_ptrace_data_t;
#define R_PTRACE_NODATA 0
#endif
```
```C
#define debug_read_raw(io,x,y) r_io_ptrace((io), PTRACE_PEEKTEXT, (x), (void *)(y), R_PTRACE_NODATA)
```
and avoid all together the new `#if defined...` you are introducing in this PR. This way things will be more "declarative" and there will be less `#if..` to follow.

Comment (devnexen) [reply libr/io/p/io_ptrace.c]: Ok did a PR for this little code simplification.

Comment (ret2libc) [reply libr/io/p/io_ptrace.c]: (ops I didn't see the PR was already merged...)

Comment (devnexen) [reply libr/io/p/io_ptrace.c]: That s ok there is no functional change nor bug introduction ;-)

**Pull request merged**